src/main/kotlin/com/openlattice/indexing/BackgroundIndexingService.kt

```diff
@@ -163,21 +163,21 @@ class BackgroundIndexingService(
         }, Function<ResultSet, UUID> { ResultSetAdapters.id(it) })
     }
 
-    private fun getPropertyTypeForEntityType(entityTypeId: UUID): Map<UUID, PropertyType> {
+    internal fun getPropertyTypeForEntityType(entityTypeId: UUID): Map<UUID, PropertyType> {
         return propertyTypes
                 .getAll(entityTypes[entityTypeId]?.properties ?: setOf())
                 .filter { it.value.datatype != EdmPrimitiveTypeKind.Binary }
     }
 
-    private fun indexEntitySet(entitySet: EntitySet): Int {
+    internal fun indexEntitySet(entitySet: EntitySet, entityKeyIds: Optional<Iterable<UUID>> = Optional.empty()): Int {
```

Comment:
Why do we have this extra argument if we only use it with empty case?

Reply:
Good catch. Fixed as what we really needed was a boolean.